compiler/optimizer/VPConstraint.cpp

```diff
@@ -1293,6 +1293,8 @@ TR::VPKnownObject *TR::VPKnownObject::create(OMR::ValuePropagation *vp, TR::Know
    {
    TR::KnownObjectTable *knot = vp->comp()->getKnownObjectTable();
    TR_ASSERT(knot, "Can't create a TR::VPKnownObject without a known-object table");
```

Comment:
In light of the newly added conditional just below, I was going to say that this assertion is now bogus. But on second thought I think maybe we should keep the assertion, and push the null test out into the callers. The caller claims to have passed us a known object index, which is nonsensical in the absence a known object table.

Reply:
It does seem odd that this function can be called when there is no KNOT, unless there's an "undefined/unknown" index value. And presumably every caller is just about to look at whether the return value is NULL...